celery/utils/iso8601.py

```diff
@@ -53,6 +55,7 @@
 
 def parse_iso8601(datestring):
     """Parse and convert ISO-8601 string to datetime."""
+    warn("parse_iso8601", "v5.3", "v6", "datetime.datetime.fromisoformat")
```

Comment:
is this raising warning properly?

Reply:
In used what I found [built in](https://github.com/celery/celery/blob/3bff3f06740a0d509f807e14702f7144b043ae54/celery/utils/deprecated.py#L23).  The parameter `depracation` is provided, so the final result is a `warnings.warn` with a CDeprecationWarning.  So yes, I believe so.